main.go

```diff
@@ -238,6 +239,8 @@ func main() {
 			WatchFilterValue: watchFilterValue,
 			WaitInfraPeriod:  waitInfraPeriod,
 			Endpoints:        awsServiceEndpoints,
+			NewOCMClient:     rosa.NewOCMClient,
```

Comment:
This is not a good idea to create the clients here in the main . why do you need to do that ? for  integration test you may just mock the clients.

Reply:
Ok, I moved the initialization to `SetupWithManager` that seems to me as the only `ROSAMachinePoolReconciler` method where it fits. I am using same mechanism to create the mock implementation as the STS api.